### DISCUSSION TITLE
How to display EffectCopyAdornerTemplate during drag

### DISCUSSION THREAD

**Discussion by kermitthegreat** [punker76/gong-wpf-dragdrop#384] [Q&A]
hi,

is it possible right now to display the above mentioned DataTemplate while dragging even though the underlying element is not a drop target?

take the attached screencast as an example please.

my code looks like this right now:

```xaml
<availableActions:AvailableWorkflowAction x:Class="SteadySuite.SteadyPrint.WorkflowUi.ActionUiWrapper.AvailableActions.AvailablePrintActionUiWrapper"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:dd="clr-namespace:GongSolutions.Wpf.DragDrop;assembly=GongSolutions.Wpf.DragDrop"
    xmlns:availableActions="clr-namespace:SteadySuite.SteadyPrint.WorkflowUi.ActionUiWrapper.AvailableActions"
    xmlns:dxlc="http://schemas.devexpress.com/winfx/2008/xaml/layoutcontrol"
    DataContext="{Binding RelativeSource={RelativeSource Self}}"
    dd:DragDrop.IsDragSource="True"
    dd:DragDrop.IsDropTarget="False">
    <dd:DragDrop.EffectMoveAdornerTemplate>
        <DataTemplate>
            <Border CornerRadius="8" Background="{Binding BackgroundColorBrush}" Margin="8">
                <Border.LayoutTransform>
                    <ScaleTransform ScaleX="0.3" ScaleY="0.3"/>
                </Border.LayoutTransform>
                <dxlc:LayoutGroup Orientation="Vertical" Margin="10">
                    <TextBlock FontSize="16" FontFamily="Segoe UI" Foreground="{Binding FontColorBrush}" Text="{Binding Caption}" VerticalAlignment="Top" Margin="0,0,0,10"/>
                    <TextBlock FontSize="11" FontFamily="Segoe UI" Foreground="{Binding FontColorBrush}" Text="{Binding Description}" VerticalAlignment="Top" TextTrimming="WordEllipsis" TextWrapping="Wrap"/>
                </dxlc:LayoutGroup>
            </Border>
        </DataTemplate>
    </dd:DragDrop.EffectMoveAdornerTemplate>
```

off topic: what is the difference and how do i differentiate between the different AdornerTemplates? (EffectCopyAdornerTemplate, DragAdornerTemplate, EffectMoveAdornerTemplate a.s.o.)

thanks in advance.

**Comment by kermitthegreat** [comment]
looks like attaching a .zip file doesnt work for whatever reason.

please rename the attached file to .zip
![ice_video_20170328-104009](https://cloud.githubusercontent.com/assets/16137113/24396351/28b490b6-13a3-11e7-9de8-4b1fb98b609a.jpg)

**Comment by kermitthegreat** [comment]
@punker76 any comments on this would be helpful, so i can evaluate for our upcoming release if this is going to work or not :)

thanks in advance.

btw a great framework. greetings from siegen ;)

**Comment by punker76** [comment]
@kermitthegreat I'll give you some feedback very soon. Sorry for delay.

**Comment by mangoyoga80** [comment]
@punker76  Do you have any news in this issue of display the (default) drag adorner even when dragging over areas that don't accept drop?

**Comment by punker76** [comment]
@mangoyoga80 The DragAdorner of the item is now always visible (v2.4.0). Only the effect will be changed regarding to the target.

>off topic: what is the difference and how do i differentiate between the different AdornerTemplates? (EffectCopyAdornerTemplate, DragAdornerTemplate, EffectMoveAdornerTemplate a.s.o.)

@kermitthegreat Documentation of the properties now available here https://github.com/punker76/gong-wpf-dragdrop/wiki/Properties

>is it possible right now to display the above mentioned DataTemplate while dragging even though the underlying element is not a drop target?

Yes, if you use the `DragAdornerTemplate` instead one of the effect templates.
